### ISSUE TITLE
Pipe naming convention is not explicitly described

### ISSUE THREAD

**Issue by amayer42** [angular/angular#31787]
# 📚 Docs or angular.io bug report

### Description

It is not currently clear in the documentation how to name pipes.

### What's the affected URL?

https://angular.io/guide/styleguide#style-02-09

### Reproduction Steps

Naming and using a pipe with the `name` "my-pipe" will cause issues at runtime.  This is the case for any pipe whose `name` property is written in kebab case.

### Expected vs Actual Behavior
I would expect that the documentation explicitly state that camel case is the proper format for pipes.  Currently, the example shown is given in camel case, however, there is nothing stating that kebab case is invalid.

## 📷Screenshot

![image](https://user-images.githubusercontent.com/5626443/61659701-2063e400-ac96-11e9-908d-71f5c56fdb68.png)

## 🔥 Exception or Error
<pre><code>
Template parse errors:
Parser Error: Unexpected token '-' at
</code></pre>

See here for reproduction: https://stackblitz.com/edit/angular-issue-pipe-with-dash

**Comment by Airblader** [comment]
I think this should either be made to work or it should be a compile time error.

**Comment by amayer42** [comment]
I will note that when I ran this locally, VS code showed me the error (though it was the same sort of unintelligible error that I gave above).  I assume that means that the Angular Language Service is actually detecting it it as an issue.

Regardless of whether it is made to work, I'd still argue that the documentation be made explicit as to what naming conventions are supported.

**Comment by santoshyadav198613** [comment]
Agreed with @Airblader, @kapunahelewong can we get some inputs here.

**Comment by santoshyadav198613** [comment]
![image](https://user-images.githubusercontent.com/11923975/61691939-4ea2fb80-ad4a-11e9-9a6d-3d88cef00817.png)
So this is what i have found, the template splits the text in interpolation based on few chars and "-" is one of them.

**Comment by Airblader** [comment]
I guess that makes sense because it interprets it as a subtraction.

**Comment by amayer42** [comment]
Out of curiosity, does it make sense to additionally add this to codelyzer?  There are rules for both component-selector and directive-selector.  It seems like the same sort of thing should fit in there.  Thoughts?

**Comment by jbogarthyde** [comment]
Resolved by https://github.com/angular/angular/pull/31806
Leaving open for response to question about codelyzer.

**Comment by mgechev** [comment]
I'd suggest to implement this as part of the ESLint Angular effort.
